src/test/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperTests.java

```diff
@@ -1049,6 +1050,23 @@ public void testBuilder_whenBinaryWithLegacyKNNEnabled_thenException() {
         assertTrue(ex.getMessage(), ex.getMessage().contains("is not supported for"));
     }
 
+    public void testBuild_whenInvalidCharsInFieldName_thenThrowException() {
+        for (char disallowChar : Strings.INVALID_FILENAME_CHARS) {
+            // When an invalid vector field name was given.
+            final String invalidVectorFieldName = "fieldname" + disallowChar;
+
+            // Prepare context.
+            Settings settings = Settings.builder().put(settings(CURRENT).build()).put(KNN_INDEX, true).build();
+            Mapper.BuilderContext builderContext = new Mapper.BuilderContext(settings, new ContentPath());
+
+            // IllegalArgumentException should be thrown.
+            Exception e = assertThrows(IllegalArgumentException.class, () -> {
+                new KNNVectorFieldMapper.Builder(invalidVectorFieldName, null, CURRENT).build(builderContext);
+            });
+            assertTrue(e.getMessage().contains("Vector field name must not include"));
```

Comment:
nit. print `e.getMessage()` so that we know what is the original message.
```suggestion
            assertTrue(e.getMessage(), e.getMessage().contains("Vector field name must not include"));
```

Reply:
Sounds good! Will update it shortly